src/Communication/Connection.php

```diff
@@ -106,6 +106,41 @@ public function __construct($socketClient, LoggerInterface $logger = null, int $
 
         // create socket client
         if (\is_string($socketClient)) {
+
+            $parsedUrl = parse_url($socketClient);
+            if (isset($parsedUrl['scheme']) && ($parsedUrl['scheme']==='http' or $parsedUrl['scheme']==='https'))
+            {
+
+                $configURL = $socketClient.'/json/version';
+
+                $curl = curl_init($configURL);
+                curl_setopt($curl, CURLOPT_URL, $configURL);
+                curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+                $resp = curl_exec($curl);
+
+                if ($resp === false) {
+                    $err = curl_error($curl);
+                    curl_close($curl);
+
+                    throw new \Exception("Unable to request $configURL");
+                }
+                else {
```

Comment:
This else isn't doing anything. The throw in the previous if will stop the execution on this method.